katdal/test/conftest.py

```diff
@@ -0,0 +1,38 @@
+import pytest
+
+TEST_DURATION_TOLERANCE = 0.1
+
+
+def pytest_addoption(parser):
+    parser.addoption(
+        "--check-durations",
+        action="store_true",
+        help="Verify how long some tests run (the ones with an `expected_duration` mark)",
+    )
```

Comment:
At the risk of disappearing down the rabbit-hole of improvements, is it worth making TEST_DURATION_TOLERANCE settable from the command line too?

Reply:
:joy:

The typically overshoot on my laptop is 0.02 to 0.45 seconds. I even started with 0.05, then 0.06 and finally settled on 0.1. It runs 10x without failing, which makes it OK, I guess :-)

My idea with this is that I'd run `--check-durations` every now and then on my laptop, and just keep running it until all tests pass, instead of relaxing the tolerance.